pr-deploy/test/app.test.ts

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// mock unzipAndMove dependency before importing '../src/app'
+jest.mock('../src/zipper', () => {
+  return {
+    unzipAndMove: jest.fn().mockReturnValue(Promise.resolve('gs://serving-bucket/prId')),
+  };
+});
+
+import {Application} from 'probot';
+import {prDeployAppFn} from '../src/app';
+import express from 'express';
+import request from 'supertest';
+import Webhooks, {
+  WebhookPayloadPullRequest,
+  WebhookPayloadCheckRun} from '@octokit/webhooks';
+
+describe('test pr deploy app', () => {
+  let app: Application;
+  let github: any;
+  let server: express.Application;
+
+  beforeEach(() => {
+    github = {
+      checks: {
+        create: jest.fn().mockReturnValue(Promise.resolve()),
+        update: jest.fn().mockReturnValue(Promise.resolve()),
+        listForRef: jest.fn().mockImplementation(
+          params => {
+            if (params.repo == 'newRepo') {
```

Comment:
Adding "if" statements in tests makes them much more difficult to verify in code review, because the reviewer has to simulate the runtime inside their head

My suggestion is that in the `beforeEach` just set `listForRef: jest.fn(),` without the implementation. Then in each individual test case do `github.listForRef.mockReturnValue(…)` and at the end of those tests `expect(github.listForRef).toHaveBeenCalledWith(expect.objectContaining({repo: 'newRepo'}))` (or `toHaveBeenCalledWith({repo: 'newRepo'}) if that's the whole `params` object, or whatever else is relevant for that function)

Reply:
Makes sense :)